### PR TITLE
feat(version-select): Create `VersionSelect`

### PR DESCRIPTION
**Pull request by thiskevinwang**
🎟️ [Asana Task](https://app.asana.com/0/1100423001970639/1200934847586304/f)
🔍 [Preview Link](https://react-components-git-kevin-version-select-hashicorp.vercel.app/)

---

## Description

This adds a `<VersionSelect>` component for navigating to and from `latest` and versioned docs

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/version-select/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@hashicorp/version-select",
+  "description": "A component to handle selecting a version of docs to navigate to.",
+  "version": "0.1.0",
```
Comment (jescalan) [inline packages/version-select/package.json:4]: Might want to start this at `0.0.0` so that it kicks off its life with the first increment!

**Review by jescalan — APPROVED** [review]
👏 super impressive work here kevin! just a couple small things before merge:

- [x] add a license file
- [ ] remove the `package-lock` file (zach added a note in the readme on how to install local deps in a workspaces monorepo)

File: packages/version-select/package.json (hunk 1 above)
Comment (thiskevinwang) [reply packages/version-select/package.json]: updated to `0.0.0`!

**Comment by thiskevinwang** [comment]
@jescalan 
> - [ ]  add a license file

- [x] Added `LICENSE`

> - [ ] remove the package-lock file (zach added a note in the readme on how to install local deps in a workspaces monorepo)

I added `ts-jest` at the top level so the diff in this PR is of the root `package-lock.json`
There's no nested `./packages/version-select/package-lock.json`

**Comment by jescalan** [comment]
Ohhh haha ok never mind then, sorry I missed that one!

**Pull request merged**